spinalcordtoolbox/image.py

```diff
@@ -1704,3 +1707,66 @@ def compute_cross_corr_3d(image: Image, coord, xrange=list(range(-10, 10)), xshi
     # Change adjust rl_coord
     logger.info('R-L coordinate adjusted from %s to  %s)', x, x + xrange[ind_peak])
     return x + xrange[ind_peak]
+
+
+def stitch_images(fnames_in: list, fname_out='stitched.nii.gz'):
+    """
+    Stitch two (or more) images utilizing the C++-precompiled binaries of Biomedia-MIRA's stitching toolkit
+    (https://github.com/biomedia-mira/stitching) by placing a system call.
+
+    :param fnames_in: list of filenames of scans in RPI orientation. e.g. [stack1.nii.gz, stack2.nii.gz]
+    :param fname_out: filename for stitched output scan.
+    :param directory: path to temp directory where we store the result of stitching.nii.gz
+    :return: none
+    """
+    # stringify the fname list to parsable cmd parameter
+    fnames_cmd = " ".join(fnames_in)
+
+    cmd = ['isct_stitching', '-i', fnames_cmd, '-o', fname_out, '-a']
+    status, output = run_proc(cmd, verbose='verbose', is_sct_binary=True)
+    print(status, output)
+    if status != 0:
+        raise RuntimeError(f"Subprocess call {cmd} returned non-zero: {output}")
+
+    return Image(fname_out)
+
+
+def generate_stitched_qc_images(ims_in: Sequence[Image], im_out: Image):
+    """
+    Pad input and output images, so that a QC report can compare between the output of
+    'stitching', vs. the naive output of concatenating the input images together.
+    """
+    # find the max shape of all input images
+    shape_max = [max(im.data.shape[0] for im in ims_in),
+                 max(im.data.shape[1] for im in ims_in),
+                 max(im.data.shape[2] for im in ims_in)]
+
+    # pad any input images that are smaller than the max shape
+    # (the stitching tool can handle mismatched image shapes natively, but we have to manage it ourselves)
+    for im in ims_in:
+        for i in [0, 1]:
+            diff = shape_max[i] - im.data.shape[i]
+            if diff:
+                padding = [[0, 0], [0, 0], [0, 0]]
+                padding[i][0] = round(diff / 2)
+                padding[i][1] = diff // 2
+                im.data = np.pad(im.data, padding)
+
+    # create a 1-voxel blank image for visual clarity
+    im_blank = Image([shape_max[0], shape_max[1], 1])
+
+    # create a naively-stitched (RPI) image for comparison in QC report
+    # NB: we reverse the list of images because numpy's origin location (bottom) is are different than nibabel's (top)
```

Comment:
The comments in this function are really great! They basically answer all the questions I had while I was reading it.